scripts/calc_elf_addresses.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+file=$1
+text_addr=$2
+
+# We need to take addresses of .text, .data, .rodata and .bss and calculate
+# offsets, because the linker will keep everything between .data and .rodata and
+# between .rodata and .bss in the final image too, when we specify section
+# addresses. So just taking sizes of .text, .data and .rodata isn't enough.
+# We use file offset from the objdump command output to avoid unused sections.
+
+source <(
+objdump -h $file | awk '/ .text / {print "text="toupper($6)} / .rodata / {print "rodata="toupper($6)} / .data / {print "data="toupper($6)}'
+)
```

Comment:
Please no parsing and arithmetic in shell script... this is really trivial in Python! Demo, copy/paste this in your terminal:

```
python -c "from elftools.elf.elffile import ELFFile
elf=ELFFile(open('zephyr.elf','rb'))
print(hex(elf.get_section_by_name('.text').header.sh_offset))"
```
=> 0xe400

Plenty more samples in:
- git -C zephyr/ grep ELFFile
- https://github.com/eliben/pyelftools/tree/master/examples

Reply:
> git -C zephyr/ grep ELFFile


https://docs.zephyrproject.org/latest/build/cmake/index.html#supporting-scripts-and-tools

Reply:
@marc-hb https://github.com/thesofproject/sof/pull/8180/commits/072875bf3f8170d6795ea92239befc32d81025c0#diff-737b4acbcc477942bbd9b68cf553004b518f9d9d7ed8938d08f839ec3a240da3